### PR TITLE
save the slack workspace and choose a workspace in the context menu

### LINKED ISSUE #29: save the slack workspace and choose a workspace in the context menu
**Is your feature request related to a problem? Please describe.**
No, Just a new feature. (template mismatch?)

~**Describe the solution you'd like**~
~A clear and concise description of what you want to happen.~

~**Describe alternatives you've considered**~
~A clear and concise description of any alternative solutions or features you've considered.~

~**Additional context**~
~Add any other context or screenshots about the feature request here.~

I already implemented this feature.
Once the user enters slack workspace, it will be saved and displayed in the context menu.

### PR DESCRIPTION
**Pull request by uneco**
Fixes #29

- [x] `npm run lint` succeeds.
- [x] Appropriate changes to README are included in PR.

### CODE REVIEW

File: app/scripts/background.ts [hunk 1]
```diff
@@ -34,24 +30,59 @@ const onClickContextMenus = async (
       type: 'basic',
       title: chrome.i18n.getMessage('registrationSuccessTitle'),
       message: chrome.i18n.getMessage('registrationSuccessBody', [emojiName, teamName]),
-      iconUrl: info.srcUrl
+      iconUrl: imageUrl,
     })
+    return { teamName, emojiName, imageUrl, sessionInfo }
```
Comment (takanakahiko) [inline app/scripts/background.ts]: Please move this line to after `try/catch`.

This catch statement is for non-chrome browsers.
So, Communication with Slack is successful in `catch` statement too.

File: app/scripts/background.ts [hunk 2]
```diff
@@ -3,26 +3,22 @@
 
 import { getSessionInfo, openLoginForm, uploadEmoji } from './sub_modules/slack'
 
-const onClickContextMenus = async (
-  info: chrome.contextMenus.OnClickData,
-  tab: chrome.tabs.Tab,
-) => {
-  if (!info.srcUrl) {
+const addEmojiToTeam = async (imageUrl: string | undefined, givenTeamName?: string) => {
```
Comment (takanakahiko) [inline app/scripts/background.ts]: Please specify the type of return value.

File: app/scripts/background.ts (hunk 1 above)
Comment (uneco) [reply app/scripts/background.ts]: I had misunderstood.

**Comment by takanakahiko** [comment]
Also, please edit `How to use`.
https://github.com/takanakahiko/slack-emoji-meister/blob/master/docs/index.md

File: app/scripts/background.ts (hunk 1 above)
Comment (uneco) [reply app/scripts/background.ts]: fixed.

File: app/scripts/background.ts (hunk 2 above)
Comment (uneco) [reply app/scripts/background.ts]: fixed.

**Comment by uneco** [comment]
@takanakahiko Could you review it again?

**Pull request merged**